### PR TITLE
feat: Add upgradeable proxy to ArrangerConduit (SC-41)

### PR DESCRIPTION
**Pull request by lucas-manuel**
PR into PoC branch that includes upgradeability.

### CODE REVIEW

File: src/ArrangerConduit.sol [hunk 1]
```diff
@@ -20,15 +22,12 @@ interface RegistryLike {
 // TODO: Use ERC20Helper
 // TODO: Figure out optimal way to structure natspec
 
-contract ArrangerConduit is IArrangerConduit {
+contract ArrangerConduit is IArrangerConduit, UpgradeableProxied {
```
Comment (hexonaut) [inline src/ArrangerConduit.sol:25]: In this case there is no storage in `IArrangerConduit`, but it's better practise to always put the required proxy storage at the beginning of the inheritance chain just to minimize the chance of screw up later.

File: test/AuthFunctions.t.sol [hunk 2]
```diff
@@ -8,49 +8,49 @@ import { ConduitTestBase } from "./ConduitTestBase.t.sol";
 contract Conduit_RelyTests is ConduitTestBase {
 
     function test_rely_no_auth() public {
-        vm.expectRevert("ArrangerConduit/not-authorized");
+        vm.expectRevert("UpgradeableProxy/not-authed");
```
Comment (hexonaut) [inline test/AuthFunctions.t.sol:11]: Needs to be updated to `not-authorized` when it's fixed in proxy repo.

**Review by hexonaut — COMMENTED** [review]
Minor changes and the error message needs to be fixed on proxy repo. Good to merge after that. I just reviewed the actual smart contract, but once this is in final state will do full review of tests in the parent branch when this is merged.

File: src/ArrangerConduit.sol (hunk 1 above)
Comment (lucas-manuel) [reply src/ArrangerConduit.sol:25]: Good call, updated

**Pull request merged**